pandas/core/index.py

```diff
@@ -1185,8 +1185,11 @@ def __add__(self, other):
         if isinstance(other, Index):
             return self.union(other)
         return Index(np.array(self) + other)
+
+    def __radd__(self, other):
+        return Index(other + np.array(self))
+
     __iadd__ = __add__
-    __radd__ = __add__
 
```

Comment:
note that simply having `__radd__ = __add__` won't make the examples in the unit test work, because additions on strings are not commutative (obviously)